lua/bufferline/render.lua

```diff
@@ -169,7 +170,7 @@ local function render(update_names)
       if has_icons then
         local iconChar, iconHl = get_icon(buffer_name, vim.fn.getbufvar(buffer_number, '&filetype'), status)
         local hlName = is_inactive and 'BufferInactive' or iconHl
-        iconPrefix = hlName and hl(hlName) or namePrefix
+        iconPrefix = has_icon_custom_colors and hl('Buffer' .. status .. 'Icon') or hlName and hl(hlName) or namePrefix
```

Comment:
Instead of defining highlight groups, there's also the simpler alternative of having the icons follow the `namePrefix` when `icon_custom_colors` is set:
```lua
iconPrefix = has_icon_custom_colors and namePrefix or hlName and hl(hlName) or namePrefix
```
I like the simplicity in this, but others might like having the ability to set the colors individually.